.github/workflows/test.yml

```diff
@@ -0,0 +1,25 @@
+name: test
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
```

Comment:
FWIW this is already the default: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request